params/config.go

```diff
@@ -721,6 +728,47 @@ func (c *ChainConfig) LatestFork(time uint64) forks.Fork {
 	}
 }
 
+func (c *ChainConfig) LatestFork(num *big.Int, time uint64) forks.Fork {
+	switch {
+	case c.IsPrague(num, time):
+		return forks.Prague
+	case c.IsCancun(num, time):
+		return forks.Cancun
+	case c.IsShanghai(num, time):
+		return forks.Shanghai
+	case c.IsParis(num):
+		return forks.Paris
+	case c.IsGrayGlacier(num):
+		return forks.GrayGlacier
+	case c.IsArrowGlacier(num):
+		return forks.ArrowGlacier
+	case c.IsLondon(num):
+		return forks.London
+	case c.IsBerlin(num):
+		return forks.Berlin
+	case c.IsMuirGlacier(num):
+		return forks.MuirGlacier
+	case c.IsIstanbul(num):
+		return forks.Istanbul
+	case c.IsPetersburg(num):
+		return forks.Petersburg
+	case c.IsConstantinople(num):
+		return forks.Constantinople
+	case c.IsByzantium(num):
+		return forks.Byzantium
+	case c.IsEIP158(num):
+		return forks.SpuriousDragon
+	case c.IsEIP155(num):
+		return forks.TangerineWhistle
+	case c.IsEIP150(num):
+		return forks.DAO
```

Comment:
SO, on mainnet: 
- EIP-155 and EIP-158 were both applied in the _same fork_ at block `2_675_000`. 
- EIP-150 was applied at block `2_463_000`
- Dao was applied at block `1_920_000`. 


So this switch here is _not_ correct

Reply:
fixed